server/src/main/java/org/elasticsearch/action/update/UpdateRequest.java

```diff
@@ -800,7 +772,6 @@ public void readFrom(StreamInput in) throws IOException {
             doc = new IndexRequest();
             doc.readFrom(in);
         }
-        fields = in.readOptionalStringArray();
```

Comment:
So removing this line is going to cause problems in mixed clusters. Elasticsearch versions before 7.0.0 will continue to send this array. You have to read it and decide what to do with it. Something like:
```
        if (false == in.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
            String[] fields = in.readOptionalStringArray();
            if (fields != null) {
                throw new IllegalArgumentException("[fields] is no longer supported");
            }
        }
```

is probably fine to be honest.